BEIMA.Backend/ReportService/ReportWriter.cs

```diff
@@ -183,6 +247,68 @@ public static byte[] GenerateAllDeviceReports(List<DeviceType> deviceTypes, List
             return zipFileBytes;
         }
 
+        /// <summary>
+        /// Creates a list of strings filled with all of the property names in the DeviceType object
+        /// excluding the Fields properties. In addition another header is included for the count
+        /// of devices associated with a device type
+        /// </summary>
+        /// <returns>List of headers for a device type object</returns>
+        private static List<string> GenerateDeviceTypeReportHeaders()
+        {
+            var headers = new List<string>();
+
+            // Add all of the general prop names associated with the Device object
+            var generalProps = DeviceTypeReportProps.GeneralProps.Select(p => p.Name);
+            headers.AddRange(generalProps);
+
+            // Add all of the last modified values contained in the deviceType
+            var lastModifiedProps = DeviceTypeReportProps.LastModifiedProps.OrderBy(val => val.Key).Select(val => val.Value);
+            headers.AddRange(lastModifiedProps);
+
+            // Add header for count of all devices associated with the device type
+            headers.Add("DeviceCount");
+
+            return headers;
+        }
+
+        /// <summary>
+        /// Creates a list of strings filled with all of the property values in a DeviceType object. In addition
+        /// another value is added for the count of the number of devices associated with the DeviceType.
+        /// </summary>
+        /// <param name="deviceType">DeviceType that the values should be based on</param>
+        /// <param name="associatedDevices">List of devices associated with the DeviceType</param>
+        /// <returns></returns>
+        private static List<string> GenerateDeviceTypeReportValues(DeviceType deviceType, List<Device> associatedDevices)
+        {
```

Comment:
Feels bad not including the custom fields. Maybe we could put all of the custom fields into a single cell for each device type? Idk, just feels like we should be including them.

Reply:
I'm somewhat hesitant to do so just because csv's whose rows don't have the same amount of columns tend to be pretty messy and hard to look at.

Reply:
I was thinking that we would use a different dilimeter than a comma for that cell, so that they are all in the same box, but I could see that being a bit messy as well. I'm just wondering if there is any reasonable way for us to include them.